### DISCUSSION TITLE
Utilizing validators library to have sanity check on entities

### DISCUSSION THREAD

**Discussion by ogencoglu** [microsoft/presidio#1290] [Ideas]
[validators](https://python-validators.github.io/validators/) library provides predefined validators for all sorts of things such as email, url, social security numbers, credit card numbers, iban etc.

Would it make sense to utilize this library to have a final post-processing sanity filter?

Also probably would be interesting to check if `presidio` vs `validators` implementations differ regarding email recognition etc.

**Comment by omri374** [comment]
If I understand correcrly, `validators` would work only on the substring which contains the value, right? So we would have to extract the value and pass it to validators.

If that is the case, I think it makes sense to use it for specific pii entities validation, either for new entities or if their validation is better thqn Presidio's. If it also detects the entities themselves, it would be great to add it as an extension for additional types of entities Presidio does not support.

**Comment by ogencoglu** [comment]
> If I understand correcrly, validators would work only on the substring which contains the value, right? So we would have to extract the value and pass it to validators.

Yes. It does not do entity detection.
